### PR TITLE
Fix snapshots manager

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- during a sync from genesis on testnet, the metachain node ends up having the PeerAccountsTrie DB at ~13GB even if the config.toml specified that `SnapshotsEnabled = false` and `PeerStatePruningEnabled = true`
The snapshotting process waited for epoch change & snapshot markup even if the trie storage manager did not perform snapshots. The result was that the memoryEvictionWaitingList started dropping stored keys so the correct state pruning was not performed.
  
## Proposed changes
- added an earlier exit point on the snapshotsManager handler

## Testing procedure
- standard system test
- start from genesis on testnet on all shards and measure the space occupied by the DBs. 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: state/snapshotsManager.go [hunk 1]
```diff
@@ -155,6 +155,15 @@ func (sm *snapshotsManager) SnapshotState(
 	epoch uint32,
 	trieStorageManager common.StorageManager,
 ) {
+	if check.IfNil(trieStorageManager) {
+		return
+	}
+	if !trieStorageManager.IsSnapshotSupported() {
+		log.Debug("skipping snapshot as the snapshot is not supported by the current trieStorageManager",
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply state/snapshotsManager.go:162]: The reason I have added the new function is that the original trieStorageManager that supports snapshotting has logic inside the `ShouldTakeSnapshot`. And that logic we want to execute after the node steps into the new epoch.

[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply state/snapshotsManager.go:162]: It won't work as it is right now. There are a few things that need to be set before `ShouldTakeSnapshot` is called if the snapshots are enabled. Will refactor in a future PR to merge these 2 funcs.

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by valentin-lup — APPROVED** [review]
Tested on Public Testnet Stack

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: master-e134e5314d -> fix-snapshots-manager-81f69cfdee

--- Specific errors ---

block hash does not match 12439
wrong nonce in block 4188
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  22
Nr. of all WARNS:  9673
Nr. of new ERRORS:  21
Nr. of new WARNS:  132
Nr. of PANICS:  0

/------/

**Pull request merged**